### ISSUE TITLE
Is this a valid operation in NLI datasets?

### ISSUE THREAD

**Issue by Punchwes** [UKPLab/sentence-transformers#947]
Hi, thanks for releasing the new training script for NLI data. I am a little bit confused about these lines

https://github.com/UKPLab/sentence-transformers/blob/25169a38676781cf60ac6db9a4823d7ac727f954/examples/training/nli/training_nli_v2.py#L70-L74

For this new script.. you added this "#Also add the opposite " thing which doesn't sound plausible to me... You are assuming the relation still holds when you change the order of these two sentences but this is not always true in natural language inference...

Did I miss anything... Or you have a particular reason to add these reversed instances?

**Comment by nreimers** [comment]
Hi @Punchwes 
The loss uses cosine similarity to measure the similarity of sentence pairs. Cosine similarity is symmetric, i.e., cossim(A, B) = cossim(B, A). Hence, we can also add the opposite pair.

In terms of the entailment relation it is correct that from A => B does not follow B => A. But as the vector space is symmetric, we cannot decide on the directions and just want to have A and B close in vector space.

**Comment by Punchwes** [comment]
Thanks for you explanation, it makes every sense. I wonder if the v2 script **_training_nli_v2.py_** is exactly the same scenario of SimCSE in supervised setting.. I have gone through the code, it looks it matches the description in their paper except different learning rate, batch size and epoch number..

**Comment by nreimers** [comment]
Hi,
Yes, the SimCSE authors use this type of training for their supervised experiments.
